### PR TITLE
Switch from hack_load_vm_file_hook to PlatformPlugin

### PR DESCRIPTION
**Pull request by abarth**
This patch switches us from using our previous hacking approach to integrating
with package:test to using the new PlatformPlugin interface.

### CODE REVIEW

**Comment by abarth** [comment]
Depends on a futuristic version of package:test

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import 'package:test/src/runner/loader.dart' as hack;
+
+final String _kSkyShell = Platform.environment['SKY_SHELL'];
+const String _kHost = '127.0.0.1';
+const String _kPath = '/runner';
+
+String shellPath;
+
+void installHook() {
+  hack.createVMPlatform = (Configuration config) => new FlutterPlatform(config);
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: I'm curious, is this API more supported, or close to what a supported API will look like?

Comment (abarth) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: No idea.  It's just what I hacked into the test package to get this running.  I'm sure the maintainers of the test package will do something cleaner.

**Comment by abarth** [comment]
The future is now.  @Hixie

**Comment by Hixie** [comment]
LGTM.

At some point we should try testing this. In particular I'm curious about making sure we correctly handle:
* crash on startup
* crash during test
* crash after test

**Pull request merged**